modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/raft/CmgRaftGroupListener.java

```diff
@@ -350,11 +329,7 @@ private void changeMetastorageNodes(ChangeMetaStorageInfoCommand command) {
 
     @Override
     public void onSnapshotSave(Path path, Consumer<Throwable> doneClo) {
-        inBusyLock(busyLock, () -> onSnapshotSaveBusy(path, doneClo));
-    }
-
-    private CompletableFuture<Void> onSnapshotSaveBusy(Path path, Consumer<Throwable> doneClo) {
-        return storageManager.snapshot(path)
+        storageManager.snapshot(path)
```

Comment:
Could you please elaborate why it's now not possible to persist incorrect appliedIndex within snapshotting?

Reply:
Because appliedIndex doesn't become invalid anymore. It could happen in the previous version of this code (when we would skip command execution, but we would not throw an exception, so FSMCaller would think that the command was applied). In this PR, we stop applying commands as soon as we see that a shutdown was requested.

Reply:
Do you mean that runTheRestClosureWithShutdownException doesn't move IteratorImpl#currentIndex?

Reply:
It doesn't move `currentIndex`